### PR TITLE
Add support for ALTER VIEW RENAME TO statement

### PR DESCRIPTION
**Pull request by dmariamgeorge**
## Description
Add support for SQL statement ALTER VIEW <view_name> RENAME TO <new_view_name>. 

## Motivation and Context

Addresses issue: https://github.com/prestodb/presto/issues/23748

## Impact

Support SQL statement ALTER VIEW <view_name> RENAME TO <new_view_name> in Iceberg connector.

## Test Plan

- Modified `testTableRules()` in `TestFileBasedAccessControl.java` to include check for rename view. 
- Added test `testRenameView()` in `TestSqlParser.java`.
- In memory connector, added tests `testRenameView()` and `testRenameViewIfNotExists()` in `TestMemorySmoke.java`.
- Modified test `testViews()` in `TestMemoryMetadata.java` to include check for rename view. 
- In Iceberg connector, added tests `testRenameView()` and `testRenameViewIfNotExists()` in `TestIcebergMetadataListing.java`. 

## Contributor checklist

- [x] Please make sure your submission complies with our [development](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#development), [formatting](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#formatting), [commit message](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#commit-formatting-and-pull-requests), and [attribution guidelines](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#attribution).
- [x] PR description addresses the issue accurately and concisely.  If the change is non-trivial, a GitHub Issue is referenced.
- [x] Documented new properties (with its default value), SQL syntax, functions, or other functionality.
- [x] If release notes are required, they follow the [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines).
- [x] Adequate tests were added if applicable.
- [x] CI passed.

## Release Notes
Please follow [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines) and fill in the release notes below.

```
== RELEASE NOTES ==
General Changes
* Add support for ``ALTER VIEW RENAME TO`` operation, including the necessary infrastructure for connector implementations. :pr:`23749`

Iceberg Connector Changes
* Add support for ``ALTER VIEW RENAME TO``. :pr:`23749`

Memory Connector Changes
* Add support for ``ALTER VIEW RENAME TO``. :pr:`23749`
```

### CODE REVIEW

**Comment by steveburnett** [comment]
Consider adding documentation for this to https://prestodb.io/docs/current/connector/iceberg.html#sql-support.

**Review by steveburnett — DISMISSED** [review]
LGTM! (docs)

Pull branch, local doc build, looks good. Thanks!

**Comment by agrawalreetika** [comment]
Thanks for the PR.

I have a few suggestions -
- I think we can add one commit about adding SQL statement support and different commits for support for `ALTER VIEW RENAME` support for each connector support
- Please add tests as well in respective connector support commits
- We can add `ALTER VIEW` documentation along with 1st commit of SQL Statement support here https://prestodb.io/docs/current/sql.html
- Also it would be good to add SQL statement support for `ALTER VIEW [ IF EXISTS ]` as well and its respective tests

File: presto-docs/src/main/sphinx/connector/iceberg.rst [hunk 1]
```diff
@@ -1190,6 +1190,12 @@ The table is partitioned by the transformed value of the column::
 
      ALTER TABLE iceberg.web.page_views ADD COLUMN ts timestamp WITH (partitioning = 'hour');
 
+ALTER VIEW
```
Comment (hantangwangd) [inline presto-docs/src/main/sphinx/connector/iceberg.rst:1193]:
```suggestion
ALTER VIEW
^^^^^^^^^^
```

File: presto-docs/src/main/sphinx/connector/iceberg.rst [hunk 2]
```diff
@@ -1190,6 +1190,12 @@ The table is partitioned by the transformed value of the column::
...
+
+Alter view operations to alter the name of an existing view to a new name is supported in the Iceberg connector::
```
Comment (hantangwangd) [inline presto-docs/src/main/sphinx/connector/iceberg.rst:1195]: Should we clarify that we only supports altering views in Hive and Glue metastores? The same as `CREATE VIEW` above.

File: presto-docs/src/main/sphinx/connector/iceberg.rst [hunk 3]
```diff
@@ -1190,6 +1190,12 @@ The table is partitioned by the transformed value of the column::
...
+
+     ALTER TABLE iceberg.web.page_views RENAME TO iceberg.web.page_new_views;
```
Comment (hantangwangd) [inline presto-docs/src/main/sphinx/connector/iceberg.rst:1197]:
```suggestion
     ALTER VIEW iceberg.web.page_views RENAME TO iceberg.web.page_new_views;
```

File: presto-main/src/main/java/com/facebook/presto/execution/RenameViewTask.java [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+import static java.util.Objects.requireNonNull;
+
+public class RenameViewTask
+        implements DDLDefinitionTask<RenameView>
+{
+    private final Metadata metadata;
+    private final AccessControl accessControl;
+
+    @Inject
+    public RenameViewTask(Metadata metadata, AccessControl accessControl)
```
Comment (hantangwangd) [inline presto-main/src/main/java/com/facebook/presto/execution/RenameViewTask.java:47]: Is it necessary to accept `metadata` and `accessControl` here?

File: presto-main/src/main/java/com/facebook/presto/execution/RenameViewTask.java [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+    @Override
+    public String getName()
+    {
+        return "RENAME VIEW";
+    }
+
+    public ListenableFuture<?> execute(RenameView statement, TransactionManager transactionManager, Metadata metadata, AccessControl accessControl, Session session, List<Expression> parameters, WarningCollector warningCollector)
+    {
+        QualifiedObjectName name = createQualifiedObjectName(session, statement, statement.getSource());
+        QualifiedObjectName viewName = createQualifiedObjectName(session, statement, statement.getSource());
```
Comment (hantangwangd) [inline presto-main/src/main/java/com/facebook/presto/execution/RenameViewTask.java:61]: Why declare two variables for the same view name here?

File: presto-main/src/main/java/com/facebook/presto/testing/LocalQueryRunner.java [hunk 6]
```diff
@@ -566,6 +568,7 @@ private LocalQueryRunner(Session defaultSession, FeaturesConfig featuresConfig,
                 .put(DropMaterializedView.class, new DropMaterializedViewTask())
                 .put(RenameColumn.class, new RenameColumnTask())
                 .put(RenameTable.class, new RenameTableTask())
+                .put(RenameView.class, new RenameViewTask(metadata, accessControl))
```
Comment (hantangwangd) [inline presto-main/src/main/java/com/facebook/presto/testing/LocalQueryRunner.java:571]: The same as mentioned above, is it necessary to pass `metadata` and `accessControl` on construction of `RenameViewTask`?

**Review by hantangwangd — CHANGES_REQUESTED** [review]

**Comment by ZacBlanco** [comment]
@dmariamgeorge Can we split this into two commits like @agrawalreetika suggested? Once we have both commits I will also review

**Comment by steveburnett** [comment]
Minor nits of formatting suggested for the release note entry.

```
== RELEASE NOTES ==
General Changes
* Add support for ``ALTER VIEW RENAME TO``. :pr:`23749`

Iceberg Connector Changes
* Add support for ``ALTER VIEW RENAME TO``. :pr:`23749`
```

**Comment by dmariamgeorge** [comment]
> @dmariamgeorge Can we split this into two commits like @agrawalreetika suggested? Once we have both commits I will also review

Sure @ZacBlanco. Working on the same.

**Comment by dmariamgeorge** [comment]
> > @dmariamgeorge Can we split this into two commits like @agrawalreetika suggested? Once we have both commits I will also review
> 
> Sure @ZacBlanco. Working on the same.

@ZacBlanco - I have split the commits as suggested by @agrawalreetika. Can you please take a look?

File: presto-iceberg/src/test/java/com/facebook/presto/iceberg/TestIcebergMetadataListing.java [hunk 7]
```diff
@@ -153,4 +153,16 @@ public void testTableValidation()
         assertQuerySucceeds("SELECT * FROM iceberg.test_schema.iceberg_table1");
         assertQueryFails("SELECT * FROM iceberg.test_schema.hive_table", "Not an Iceberg table: test_schema.hive_table");
     }
+
+    @Test
+    public void testRenameView()
+    {
+        assertQuerySucceeds("CREATE TABLE iceberg.test_schema.iceberg_test_table (_string VARCHAR, _integer INTEGER)");
+        assertUpdate("CREATE VIEW iceberg.test_schema.test_view_to_be_renamed AS SELECT * FROM iceberg.test_schema.iceberg_test_table");
+        assertUpdate("ALTER VIEW IF EXISTS iceberg.test_schema.test_view_to_be_renamed RENAME TO iceberg.test_schema.test_view_renamed");
```
Comment (agrawalreetika) [inline presto-iceberg/src/test/java/com/facebook/presto/iceberg/TestIcebergMetadataListing.java:162]: Can you please add 2 more test here when table doesn't exist. One with `IF EXISTS` and one without it?

File: presto-docs/src/main/sphinx/sql/alter-view.rst [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+and recreate it. The view's definition, security settings, and dependencies
+remain unchanged; only the name of the view is updated.
+
+The optional ``IF EXISTS`` clause prevents an error from being raised if the
+view does not exist. Instead, no action is taken, and a notice is issued.
+
+Renaming a view does not affect the data or structure of the underlying
+query used to define the view. Any permissions or dependencies on the
+view are retained, and queries or applications using the old name will
+need to be updated to reference.
```
Comment (agrawalreetika) [inline presto-docs/src/main/sphinx/sql/alter-view.rst:26]: Lets add an Example section as well here, ref -  https://prestodb.io/docs/current/sql/drop-view.html#examples

File: presto-main/src/main/java/com/facebook/presto/execution/RenameViewTask.java [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+    }
+
+    public ListenableFuture<?> execute(RenameView statement, TransactionManager transactionManager, Metadata metadata, AccessControl accessControl, Session session, List<Expression> parameters, WarningCollector warningCollector)
+    {
+        QualifiedObjectName viewName = createQualifiedObjectName(session, statement, statement.getSource());
+
+        Optional<ViewDefinition> view = metadata.getMetadataResolver(session).getView(viewName);
+        if (!view.isPresent()) {
+            if (!statement.isExists()) {
+                throw new SemanticException(MISSING_TABLE, statement, "View '%s' does not exist", viewName);
```
Comment (agrawalreetika) [inline presto-main/src/main/java/com/facebook/presto/execution/RenameViewTask.java:54]: Lets add a new SemanticErrorCode for `MISSING_VIEW`

File: presto-main/src/main/java/com/facebook/presto/execution/RenameViewTask.java [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+            }
+            return immediateFuture(null);
+        }
+
+        QualifiedObjectName target = createQualifiedObjectName(session, statement, statement.getTarget());
+        if (!metadata.getCatalogHandle(session, target.getCatalogName()).isPresent()) {
+            throw new SemanticException(MISSING_CATALOG, statement, "Target catalog '%s' does not exist", target.getCatalogName());
+        }
+        if (metadata.getMetadataResolver(session).getView(target).isPresent()) {
+            throw new SemanticException(TABLE_ALREADY_EXISTS, statement, "Target view '%s' already exists", target);
```
Comment (agrawalreetika) [inline presto-main/src/main/java/com/facebook/presto/execution/RenameViewTask.java:64]: Lets add a new SemanticErrorCode for `VIEW_ALREADY_EXISTS`

File: presto-parser/src/test/java/com/facebook/presto/sql/parser/TestSqlParser.java [hunk 11]
```diff
@@ -1504,6 +1505,12 @@ public void testTruncateTable()
         assertStatement("TRUNCATE TABLE a.b.c", new TruncateTable(QualifiedName.of("a", "b", "c")));
     }
 
+    @Test
+    public void testRenameView()
+    {
+        assertStatement("ALTER VIEW a RENAME TO b", new RenameView(QualifiedName.of("a"), QualifiedName.of("b"), false));
+    }
```
Comment (agrawalreetika) [inline presto-parser/src/test/java/com/facebook/presto/sql/parser/TestSqlParser.java:1512]: Lets add a test for `IF EXISTS` as well in the Parser

**Review by agrawalreetika — CHANGES_REQUESTED** [review]
Thank you for making the changes. 
Looks like some of file changes need to be shifted in respective commits. I have dropped in some review comments please check.

**Review by agrawalreetika — DISMISSED** [review]
@dmariamgeorge Thank you for making the changes. LGTM

File: presto-docs/src/main/sphinx/connector/iceberg.rst [hunk 12]
```diff
@@ -1190,6 +1190,15 @@ The table is partitioned by the transformed value of the column::
 
      ALTER TABLE iceberg.web.page_views ADD COLUMN ts timestamp WITH (partitioning = 'hour');
 
+ALTER VIEW
+^^^^^^^^^^
+
+Alter view operations to alter the name of an existing view to a new name is supported in the Iceberg connector::
```
Comment (steveburnett) [inline presto-docs/src/main/sphinx/connector/iceberg.rst:1196]:
```suggestion
Alter view operations to alter the name of an existing view to a new name is supported in the Iceberg connector.
```

Using `::` here causes the code block to look like this in a local doc build: 
Screenshot 2024-11-13 at 10 44 22 AM

I tested this edit in a local doc build and it fixes the code block.

File: presto-docs/src/main/sphinx/connector/memory.rst [hunk 13]
```diff
@@ -92,6 +92,15 @@ To delete an existing table:
 
 .. note:: After using ``DROP TABLE``, memory is not released immediately. It is released after the next write access to the memory connector.
 
+ALTER VIEW
+^^^^^^^^^^
+
+Alter view operations to alter the name of an existing view to a new name is supported in the Memory connector::
```
Comment (steveburnett) [inline presto-docs/src/main/sphinx/connector/memory.rst:98]:
```suggestion
Alter view operations to alter the name of an existing view to a new name is supported in the Memory connector.
```
Same - using `::` here causes the code block to look like this in a local doc build: 

Screenshot 2024-11-13 at 10 48 00 AM

I tested this edit in a local doc build and it fixes the code block.

File: presto-docs/src/main/sphinx/sql/alter-view.rst [hunk 14]
```diff
@@ -0,0 +1,42 @@
+============
```
Comment (steveburnett) [inline presto-docs/src/main/sphinx/sql/alter-view.rst:1]:
```suggestion
==========
```
Formatting lines should be the same number of characters as the text being formatted.

File: presto-docs/src/main/sphinx/sql/alter-view.rst [hunk 15]
```diff
@@ -0,0 +1,42 @@
...
+ALTER VIEW
+============
```
Comment (steveburnett) [inline presto-docs/src/main/sphinx/sql/alter-view.rst:3]:
```suggestion
==========
```

**Review by steveburnett — CHANGES_REQUESTED** [review]
Thanks for the doc! Just a few formatting nits and such.

**Review by hantangwangd — CHANGES_REQUESTED** [review]
Thanks for this work, one little problem and some nits, otherwise looks good to me.

File: presto-hive/src/main/java/com/facebook/presto/hive/security/SqlStandardAccessControl.java [hunk 16]
```diff
@@ -431,6 +432,24 @@ public void checkCanCreateView(ConnectorTransactionHandle transaction, Connector
...
+    public void checkCanRenameView(ConnectorTransactionHandle transaction, ConnectorIdentity identity, AccessControlContext context, SchemaTableName viewName, SchemaTableName newViewName)
+    {
+        MetastoreContext metastoreContext = new MetastoreContext(
+                identity, context.getQueryId().getId(),
+                context.getClientInfo(),
+                context.getClientTags(),
+                context.getSource(),
+                Optional.empty(),
+                false,
+                HiveColumnConverterProvider.DEFAULT_COLUMN_CONVERTER_PROVIDER,
```
Comment (ZacBlanco) [inline presto-hive/src/main/java/com/facebook/presto/hive/security/SqlStandardAccessControl.java:445]: nit: static import

File: presto-hive/src/main/java/com/facebook/presto/hive/security/SqlStandardAccessControl.java [hunk 17]
```diff
@@ -431,6 +432,24 @@ public void checkCanCreateView(ConnectorTransactionHandle transaction, Connector
         }
     }
 
+    @Override
+    public void checkCanRenameView(ConnectorTransactionHandle transaction, ConnectorIdentity identity, AccessControlContext context, SchemaTableName viewName, SchemaTableName newViewName)
+    {
+        MetastoreContext metastoreContext = new MetastoreContext(
+                identity, context.getQueryId().getId(),
```
Comment (ZacBlanco) [inline presto-hive/src/main/java/com/facebook/presto/hive/security/SqlStandardAccessControl.java:439]: nit: split these arguments onto two lines

File: presto-hive/src/main/java/com/facebook/presto/hive/security/SystemTableAwareAccessControl.java [hunk 18]
```diff
@@ -166,6 +166,12 @@ public void checkCanCreateView(ConnectorTransactionHandle transactionHandle, Con
         delegate.checkCanCreateView(transactionHandle, identity, context, viewName);
     }
 
+    @Override
+    public void checkCanRenameView(ConnectorTransactionHandle transactionHandle, ConnectorIdentity identity, AccessControlContext context, SchemaTableName viewName, SchemaTableName newViewName)
+    {
+        delegate().checkCanRenameView(transactionHandle, identity, context, viewName, newViewName);
```
Comment (ZacBlanco) [inline presto-hive/src/main/java/com/facebook/presto/hive/security/SystemTableAwareAccessControl.java:172]: this should be `delegate`, not `delegate()`

File: presto-iceberg/src/test/java/com/facebook/presto/iceberg/TestIcebergMetadataListing.java [hunk 19]
```diff
@@ -153,4 +153,21 @@ public void testTableValidation()
...
```
Comment (ZacBlanco) [inline presto-iceberg/src/test/java/com/facebook/presto/iceberg/TestIcebergMetadataListing.java:162]: Can we add a case without the `IF EXISTS` clause?

File: presto-main/src/main/java/com/facebook/presto/execution/RenameViewTask.java [hunk 20]
```diff
@@ -0,0 +1,76 @@
...
+
+        QualifiedObjectName target = createQualifiedObjectName(session, statement, statement.getTarget());
+        if (!metadata.getCatalogHandle(session, target.getCatalogName()).isPresent()) {
+            throw new SemanticException(MISSING_CATALOG, statement, "Target catalog '%s' does not exist", target.getCatalogName());
+        }
+        if (metadata.getMetadataResolver(session).getView(target).isPresent()) {
+            throw new SemanticException(VIEW_ALREADY_EXISTS, statement, "Target view '%s' already exists", target);
+        }
+        if (!viewName.getCatalogName().equals(target.getCatalogName())) {
+            throw new SemanticException(NOT_SUPPORTED, statement, "View rename across catalogs is not supported");
```
Comment (ZacBlanco) [inline presto-main/src/main/java/com/facebook/presto/execution/RenameViewTask.java:67]: We support renaming across schemas though? Does it make sense to rename across schemas? What if the source tables don't exist in the new schema?

File: presto-memory/src/test/java/com/facebook/presto/plugin/memory/TestMemorySmoke.java [hunk 21]
```diff
@@ -185,6 +185,31 @@ public void testViews()
         assertQueryFails("DROP VIEW test_view", "line 1:1: View 'memory.default.test_view' does not exist");
     }
 
+    @Test
+    public void testRenameView()
```
Comment (ZacBlanco) [inline presto-memory/src/test/java/com/facebook/presto/plugin/memory/TestMemorySmoke.java:189]: Wondering if we should instead add these tests to `AbstractTestDistributedQueries`? This way all connectors which support views can run this and we don't need to duplicate code.

**Review by ZacBlanco — CHANGES_REQUESTED** [review]

File: presto-memory/src/test/java/com/facebook/presto/plugin/memory/TestMemorySmoke.java (hunk 21 above)
Comment (dmariamgeorge) [reply presto-memory/src/test/java/com/facebook/presto/plugin/memory/TestMemorySmoke.java:189]: @ZacBlanco - The support for ALTER VIEW RENAME TO is implemented only for the Iceberg and Memory connectors. If we add a test for this in AbstractTestDistributedQueries, it is required to override it in each connector, so the test will not fail for the other connectors.

**Review by steveburnett — DISMISSED** [review]
LGTM! (docs)

Pull updated branch, new local doc build, reviewed, looks good. Thanks!

File: presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/file/FileHiveMetastore.java [hunk 22]
```diff
@@ -511,12 +511,7 @@ public synchronized MetastoreOperationResult renameTable(MetastoreContext metast
         requireNonNull(tableName, "tableName is null");
         requireNonNull(newDatabaseName, "newDatabaseName is null");
         requireNonNull(newTableName, "newTableName is null");
-
-        Table table = getRequiredTable(metastoreContext, databaseName, tableName);
         getRequiredDatabase(metastoreContext, newDatabaseName);
-        if (isIcebergTable(table)) {
-            throw new PrestoException(NOT_SUPPORTED, "Rename not supported for Iceberg tables");
-        }
```
[the remark this answers is not in the document]
Comment (hantangwangd) [reply presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/file/FileHiveMetastore.java:519]: As I understand, it's not just for testing. When user set `hive.metastore=file` and set `hive.metastore.catalog.dir` to a local or hdfs path, they are indeed using `FileHiveMetastore` in their environment. Meanwhile, it's not a good idea to introduce additional problems in any code, even for code that is only used in testing.

Iceberg views created by hive catalog will have a table parameters `presto_view=true`, so I think we can check this parameter, let iceberg view to pass, and keep iceberg table as it is until we really support renaming iceberg table name in hive catalog. Do you think this makes sense?

[the remark this answers is not in the document]
Comment (dmariamgeorge) [reply presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/file/FileHiveMetastore.java]: Sure @hantangwangd.

**Review by hantangwangd — COMMENTED** [review]
Thanks for the fix, LGTM. Just one little nit.

File: presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/MetastoreUtil.java [hunk 23]
```diff
@@ -1026,6 +1027,11 @@ public static boolean isIcebergTable(Map<String, String> tableParameters)
         return ICEBERG_TABLE_TYPE_VALUE.equalsIgnoreCase(tableParameters.get(ICEBERG_TABLE_TYPE_NAME));
     }
 
+    public static boolean isIcebergView(Table table)
+    {
+        return "true".equalsIgnoreCase(table.getParameters().get(ICEBERG_VIEW_TYPE_NAME));
+    }
```
Comment (hantangwangd) [inline presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/MetastoreUtil.java:1033]: Let's use the already existed constant `PRESTO_VIEW_FLAG`, no need to declare another one.

**Review by hantangwangd — APPROVED** [review]
LGTM, you can rebase to the newest commit list to fix the CI test failure.

**Review by agrawalreetika — APPROVED** [review]

File: presto-docs/src/main/sphinx/sql/alter-view.rst [hunk 24]
```diff
@@ -0,0 +1,42 @@
...
+new name. This allows you to change the name of a view without having to drop
+and recreate it. The view's definition, security settings, and dependencies
+remain unchanged; only the name of the view is updated.
+
+The optional ``IF EXISTS`` clause prevents an error from being raised if the
+view does not exist. Instead, no action is taken, and a notice is issued.
+
+Renaming a view does not affect the data or structure of the underlying
+query used to define the view. Any permissions or dependencies on the
+view are retained, and queries or applications using the old name will
```
Comment (steveburnett) [inline presto-docs/src/main/sphinx/sql/alter-view.rst:25]:
```suggestion
view are retained, and queries or applications using the old name must
```

File: presto-docs/src/main/sphinx/sql/alter-view.rst [hunk 25]
```diff
@@ -0,0 +1,42 @@
...
+need to be updated to reference.
```
Comment (steveburnett) [inline presto-docs/src/main/sphinx/sql/alter-view.rst:26]:
```suggestion
be updated to use the new name.
```

**Review by steveburnett — CHANGES_REQUESTED** [review]
Thanks for the updated doc! Two minor rephrasing suggestions.

**Comment by dmariamgeorge** [comment]
Hi @ZacBlanco @agrawalreetika @steveburnett @hantangwangd 

Could you please take another look at this PR? I’ve incorporated all the review comments. Thanks!

**Review by steveburnett — APPROVED** [review]
LGTM! (docs)

Pull updated build, new local doc build, reviewed pages in local doc build. Everything looks good. Thanks!

**Review by hantangwangd — APPROVED** [review]

**Review by agrawalreetika — APPROVED** [review]

**Review by ZacBlanco — APPROVED** [review]

**Comment by ZacBlanco** [comment]
@dmariamgeorge instead of the screenshot in the PR description, can you please replace it with a short summary of the unit and integration tests that have been added

**Comment by ZacBlanco** [comment]
Also, I suggest in the release note section to be more specific in the "General Changes" section that we're adding the ability for connectors to implement `ALTER VIEW RENAME TO`. IMO It isn't clear from the two release notes that it contains the infrastructure and only an implementation for the Iceberg connector.

**Comment by dmariamgeorge** [comment]
@ZacBlanco - I have revised the PR description. Could you please review it?

**Review by ZacBlanco — APPROVED** [review]

**Comment by dmariamgeorge** [comment]
Hi @tdcmeehan 

I have received approvals from @ZacBlanco, @agrawalreetika, @hantangwangd, and @steveburnett on this PR. Could you please help merge this PR? Thanks!

**Review by tdcmeehan — APPROVED** [review]

**Pull request merged**